### PR TITLE
PIN-4221 BFF: Added Mail Id

### PR DESCRIPTION
**Pull request by nttdata-rtorsoli**

### CODE REVIEW

File: src/main/resources/interface-specification.yml [hunk 1]
```diff
@@ -11560,6 +11644,27 @@ components:
...
+        kind:
+          $ref: '#/components/schemas/MailKind'
+        address:
+          type: string
+          minLength: 5
+          maxLength: 320
+        description:
+          type: string
+          minLength: 10
+          maxLength: 250
```
Comment (galales) [inline src/main/resources/interface-specification.yml]: we should leave constraints in the process
```suggestion
          type: string
        description:
          type: string
```

File: src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/TenantsApiServiceImpl.scala [hunk 2]
```diff
@@ -218,19 +218,38 @@ final case class TenantsApiServiceImpl(
...
-    } yield ()
+    val result: Future[Unit] =
+      for {
+        tenantUuid <- tenantId.toFutureUUID
+        _          <- tenantProcessService.addTenantMail(tenantUuid, seed.toExternalModel)
+      } yield ()
+
+    onComplete(result) {
+      val headers: List[HttpHeader] = headersFromContext()
+      handleError(s"Error adding mail ${seed.address} to tenant $tenantId", headers) orElse { case Success(_) =>
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/TenantsApiServiceImpl.scala]: Potential sensitive information
```suggestion
      handleError(s"Error adding mail to tenant $tenantId", headers) orElse { case Success(_) =>
```

File: src/main/scala/it/pagopa/interop/backendforfrontend/service/impl/TenantProcessServiceImpl.scala [hunk 3]
```diff
@@ -53,14 +53,14 @@ class TenantProcessServiceImpl(tenantprocessUrl: String, blockingEc: ExecutionCo
       invoker.invoke(request, s"Revoking declared attribute $attributeId to requester Tenant")
     }
 
-  def selfcareUpsertTenant(origin: String, externalId: String, name: String)(
+  def selfcareUpsertTenant(origin: String, externalId: String, description: String)(
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/backendforfrontend/service/impl/TenantProcessServiceImpl.scala]: isn't this still the name?
```suggestion
  def selfcareUpsertTenant(origin: String, externalId: String, name: String)(
```

File: src/main/scala/it/pagopa/interop/backendforfrontend/service/impl/TenantProcessServiceImpl.scala [hunk 4]
```diff
@@ -159,4 +149,25 @@ class TenantProcessServiceImpl(tenantprocessUrl: String, blockingEc: ExecutionCo
...
       invoker.invoke(request, s"Updating verified attribute $attributeId to $tenantId")
     }
+
+  override def addTenantMail(tenantId: UUID, seed: MailSeed)(implicit contexts: Seq[(String, String)]): Future[Unit] =
+    withHeaders[Unit] { (bearerToken, correlationId) =>
+      val request: ApiRequest[Unit] =
+        api.addTenantMail(xCorrelationId = correlationId, tenantId = tenantId, mailSeed = seed)(
+          BearerToken(bearerToken)
+        )
+      invoker.invoke(request, s"Add mail ${seed.address} to $tenantId")
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/backendforfrontend/service/impl/TenantProcessServiceImpl.scala]: sensitive information
```suggestion
      invoker.invoke(request, s"Add mail to tenant $tenantId")
```

**Review by galales — APPROVED** [review]

**Review by beetlecrunch — APPROVED** [review]

**Review by galales — APPROVED** [review]

**Pull request merged**